_posts/en/newsletters/2019-11-13-newsletter.md

```diff
@@ -0,0 +1,367 @@
+  the spender if a payment attempt fails or is rejected by the receiver,
+  so routing nodes only receive income if the payment attempt succeeds.
+  However, some new applications are using this costless failure
+  mechanism to send data over LN without paying for the bandwidth they
+  use.  LN designers expected this to happen and have previously spent
+  time thinking about how they could add up-front fees to the
+  network---fees that would be paid to routing nodes whether or not a
+  payment attempt succeeded.
+
+    This week, Rusty Russell started a [thread][russell up-front] on the
+    Lightning-Dev mailing list to discuss proposals for up-front fees.
+    Russell proposed a mechanism that combines fees and hashcash-style
+    proof-of-work to try to prevent nodes from using using the extra
+    up-front payment information to guess the length of the route.
+    Anthony Towns proposed a partial [alternative][towns up-front]
+    focusing on managing payment amounts using a refund mechanism.
+
+    Joost Jager suggested that up-front payments should only be required
+    as a last resort because even small additional fees could make
+    micropayments uneconomic.  He suggested that it should be possible
+    to address bandwidth-wasting network activity using rate limits
+    based on node reputation and, further, that research into up-front
+    payments should focus on first solving liquidity abuse---where an
+    attacker ties up someone's in-channel funds for period of time---as
+    the solution to that problem may also prevent the abuse of routing
+    node bandwidth.
+
+    Ultimately, no conclusions were reached and discussion about the
+    topic remains ongoing as of this writing.
+
+- **Proposed BOLT for LN offers:** Rusty Russell [posted][bolt offers]
+  draft text for a new BOLT that would allow users to submit offers and
+  receive invoices over the LN routing protocol.  For example, Alice
+  could subscribe to a service provided by Bob where each month she
+  would submit an offer to pay Bob, Bob would reply with an invoice,
+  Alice would pay the invoice, and Bob would provide the service.
+
+    Early feedback on the proposal suggested that it might want to
+    incorporate by reference an established language for
+    machine-readable invoices, such as the [Universal Business
+    Language][] (UBL).  However, there was a concern that implementing
+    the full UBL specification would be an excessive burden on developers
+    of LN software.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[libsecp256k1][libsecp256k1 repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #16110][] adds support for building Bitcoin Core (including
+  the GUI) for the Android Native Development Kit (NDK).  In contrast to
```

Comment:
"In contrast to ... exists in the code repostitory." Consider splitting up this very long sentence.